packages-rs/drainpipe/src/db.rs

```diff
@@ -27,13 +29,20 @@ pub fn update_seq(conn: &mut SqliteConnection, new_seq: i64) -> anyhow::Result<(
 
 pub fn record_dead_letter(
     conn: &mut SqliteConnection,
+    kind: ProcessErrorKind,
     new_msg: &str,
     new_seq: i64,
+    message: &str,
 ) -> anyhow::Result<()> {
     use crate::schema::dead_letter_queue::dsl::*;
 
     diesel::insert_into(dead_letter_queue)
-        .values((msg.eq(&new_msg), seq.eq(&new_seq)))
+        .values((
+            err_kind.eq(kind as i32),
```

Comment:
Does diesel expose some `Serialize` trait we can use here instead?

Reply:
yup, there is the [ToSql](https://docs.rs/diesel/latest/diesel/serialize/trait.ToSql.html) trait, good call.  I'll implement this for `ProcessErrorKind`

i also found [this](https://kitsu.me/posts/2020_05_24_custom_types_in_diesel) if we wanted to make the database aware of the enum itself.